src/main/java/com/appdynamics/monitors/azure/AzureMonitorTask.java

```diff
@@ -8,120 +8,99 @@
 
 package com.appdynamics.monitors.azure;
 
-import com.appdynamics.extensions.conf.MonitorConfiguration;
-import com.appdynamics.monitors.azure.config.Globals;
-import com.fasterxml.jackson.databind.JsonNode;
-import com.microsoft.aad.adal4j.AuthenticationResult;
-import org.slf4j.Logger;
+import com.appdynamics.extensions.AMonitorTaskRunnable;
+import com.appdynamics.extensions.MetricWriteHelper;
+import com.appdynamics.extensions.conf.MonitorContextConfiguration;
+import com.appdynamics.extensions.util.AssertUtils;
+import com.appdynamics.monitors.azure.metrics.AzureMetrics;
+import com.appdynamics.monitors.azure.utils.Constants;
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.resources.GenericResource;
 import org.slf4j.LoggerFactory;
-import java.io.UnsupportedEncodingException;
-import java.math.BigDecimal;
-import java.net.URL;
-import java.net.URLEncoder;
-import java.text.SimpleDateFormat;
-import java.util.Calendar;
-import java.util.TimeZone;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
 
-class AzureMonitorTask implements Runnable {
-    private static final Logger logger = LoggerFactory.getLogger(AzureMonitorTask.class);
-    private final MonitorConfiguration configuration;
-    private final JsonNode node;
-    private final AuthenticationResult azureAuth;
-    private final String metric;
+@SuppressWarnings("unchecked")
+class AzureMonitorTask implements AMonitorTaskRunnable{
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(AzureMonitorTask.class);
+    private final MonitorContextConfiguration configuration;
+    private final MetricWriteHelper metricWriteHelper;
+    private final Map<String, ?> subscription;
+    private final List<Map<String,?>>  resourceGroupFilters;
+    private final CountDownLatch countDownLatch;
```

Comment:
I'm a little unclear as to why this CountDownLatch is required. I see you've created another one on line 65 that covers your Azure resources. What is the benefit of having this additional one?